ArcGISRuntimeSDKQt_CppSamples/Routing/NavigateARouteWithRerouting/NavigateARouteWithRerouting.cpp

```diff
@@ -0,0 +1,369 @@
+  });
+
+  connect(m_routeTask, &RouteTask::createDefaultParametersCompleted, this, [this](QUuid, RouteParameters defaultParameters)
+  {
+    // set values for parameters
+    defaultParameters.setReturnStops(true);
+    defaultParameters.setReturnDirections(true);
+    defaultParameters.setReturnRoutes(true);
+    defaultParameters.setOutputSpatialReference(SpatialReference::wgs84());
+
+    Stop stop1(conventionCenterPoint);
+    Stop stop2(aerospaceMuseumPoint);
+
+    QList<Stop> stopsList = {stop1, stop2};
+    defaultParameters.setStops(stopsList);
+
+    m_routeTask->solveRoute(defaultParameters);
+  });
+
+  connect(m_routeTask, &RouteTask::doneLoading, this, [this](const Error& error)
+  {
+    if (error.isEmpty())
+    {
+      m_routeTask->createDefaultParameters();
+    }
+    else
+    {
+      qDebug() << error.message() << error.additionalMessage();
+    }
+  });
+}
+
+bool NavigateARouteWithRerouting::navigationEnabled() const
+{
+  return m_navigationEnabled;
+}
+
+bool NavigateARouteWithRerouting::recenterEnabled() const
+{
+  return m_recenterEnabled;
+}
+
+void NavigateARouteWithRerouting::startNavigation()
+{
+  // disable the navigation button
+  m_navigationEnabled = false;
+  emit navigationEnabledChanged();
+
+  m_directionManeuvers = m_route.directionManeuvers(this)->directionManeuvers();
+
+  // create a route tracker
+  m_routeTracker = new RouteTracker(m_routeResult, 0, this);
+
+  connectRouteTrackerSignals();
+
+  // enable the RouteTracker to know when the QTextToSpeech engine is ready
+  m_routeTracker->setSpeechEngineReadyFunction([speaker = m_speaker]() -> bool
+  {
+    return speaker->state() == QTextToSpeech::State::Ready;
+  });
+
+  // enable "recenter" button when location display is moved from nagivation mode
+  connect(m_mapView->locationDisplay(), &LocationDisplay::autoPanModeChanged, this, [this](LocationDisplayAutoPanMode autoPanMode)
+  {
+    m_recenterEnabled = autoPanMode != LocationDisplayAutoPanMode::Navigation;
+    emit recenterEnabledChanged();
+  });
+
+  connect(m_mapView->locationDisplay(), &LocationDisplay::locationChanged, this, [this](const Location& location)
+  {
+    m_routeTracker->trackLocation(location);
+  });
+
+  // turn on map view's navigation mode
+  m_mapView->locationDisplay()->setAutoPanMode(LocationDisplayAutoPanMode::Navigation);
+  RouteTrackerLocationDataSource* dataSource = new RouteTrackerLocationDataSource(m_routeTracker, this);
+  m_mapView->locationDisplay()->setDataSource(dataSource);
+
+  ReroutingParameters* rerouteParameters = new ReroutingParameters(m_routeTask, m_routeParameters, this);
+  rerouteParameters->setStrategy(ReroutingStrategy::ToNextWaypoint);
+  rerouteParameters->setVisitFirstStopOnStart(false);
+
+  m_routeTracker->enableRerouting(rerouteParameters);
+
+  // add a data source for the location display
+  SimulationParameters* simulationParameters = new SimulationParameters(QDateTime::currentDateTime(), 40.0, 0.0, 0.0, this); // set speed
```

Comment:
to consider - these numbers in the c'tor could be split out into variables with good names, as I'm not really sure what 40.0, 0.0, and 0.0 are for without going to the API ref doc